dvc/repo/get.py

```diff
@@ -17,6 +20,20 @@
 logger = logging.getLogger(__name__)
 
 
+def copy_git_file(repo, src, dst):
+    # Assert that the file does indeed exist in git before attempting to
+    # copy it.
+    if src not in repo.scm.repo.head.commit.tree:
```

Comment:
Hm, not sure if this will work with src that has subdirs. E.g. `path/to/data`.

Reply:
i'll give it a try

Reply:
you are correct 
```
$ dvc get https://github.com/iterative/dataset-registry get-started/src/requirements.txt -o myd2
ERROR: failed to get 'get-started/src/requirements.txt' from 'https://github.com/iterative/dataset-registry' - The path 'get-started/src/requirements.txt' is outside the target repository.

```